### ISSUE TITLE
Uncaught TypeError: Cannot read property 'fonts' of undefined

### ISSUE THREAD

**Issue by vamshi9666** [storybookjs/storybook#12016]
**Describe the bug**
Ran ```yarn storybook``` after installing as per docs. And in browser console this error showed up
<img width="908" alt="Screenshot 2020-08-14 at 11 21 35 AM" src="https://user-images.githubusercontent.com/28837336/90217999-52effa00-de20-11ea-9029-a840ca0132bc.png">
 

**System:**
``` 
System:
    OS: macOS 11.0
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
  Binaries:
    Node: 13.11.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.13.7 - /usr/local/bin/npm
  Browsers:
    Chrome: 84.0.4147.125
    Firefox: 79.0
    Safari: 14.0
  npmPackages:
    @storybook/addon-actions: ^6.0.6 => 6.0.6
    @storybook/addon-essentials: ^6.0.6 => 6.0.6
    @storybook/addon-links: ^6.0.6 => 6.0.6
    @storybook/node-logger: ^6.0.6 => 6.0.6
    @storybook/preset-create-react-app: ^3.1.4 => 3.1.4
    @storybook/react: ^6.0.6 => 6.0.6
```

**Comment by schalkneethling** [comment]
@vamshi9666 I have seen this as well when trying to use `IconGallery` and `IconItem` inside a `stories.js`(seems to be fine inside `.mdx`) story. Was this your experience as well?

**Comment by vamshi9666** [comment]
I changed nothing. All I did was install and tried running `yarn storybook`

**Comment by shilman** [comment]
@vamshi9666 do you have a repro repo you can share?

**Comment by Fiszcz** [comment]
I have had the same problem.

When you remove _node_modules_ directory and then run `yarn install` or `npm install`, this errors will disappear.

But in my case, after that I had another errors:
```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (react.development.js:1431)
    at Object.useContext (react.development.js:1439)
    at useTheme (emotion-theming.browser.esm.js:112)
    at useMenu (vendors~main.fd8dd34d4e169965eb0d.bundle.js:105058)
    at Object.mapper [as current] (vendors~main.fd8dd34d4e169965eb0d.bundle.js:105491)
    at ManagerConsumer (vendors~main.fd8dd34d4e169965eb0d.bundle.js:36887)
    at renderWithHooks (vendors~main.fd8dd34d4e169965eb0d.bundle.js:129672)
    at mountIndeterminateComponent (vendors~main.fd8dd34d4e169965eb0d.bundle.js:131906)
    at beginWork$1 (vendors~main.fd8dd34d4e169965eb0d.bundle.js:133050)
    at HTMLUnknownElement.callCallback (vendors~main.fd8dd34d4e169965eb0d.bundle.js:114911)
```
```
The above error occurred in the <ManagerConsumer> component:
    in ManagerConsumer
    in Unknown (created by Layout)
    in div
    in Styled(div) (created by Sidebar)
    in Sidebar (created by Layout)
    in Layout
    in WithTheme(Layout)
    in Unknown
    in div
    in Styled(div)
    in Unknown (created by SizeMeRenderer(Component))
    in SizeMeReferenceWrapper (created by SizeMeRenderer(Component))
    in SizeMeRenderer(Component) (created by SizeMe(Component))
    in SizeMe(Component)
    in ThemeProvider
    in Unknown (created by ManagerConsumer)
    in ManagerConsumer (created by Manager)
    in EffectOnMount (created by Manager)
    in Manager (created by Context.Consumer)
    in Location (created by QueryLocation)
    in QueryLocation (created by Root)
    in LocationProvider (created by Root)
    in HelmetProvider (created by Root)
    in Root

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
```
```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (react.development.js:1431)
    at Object.useContext (react.development.js:1439)
    at useTheme (emotion-theming.browser.esm.js:112)
    at useMenu (vendors~main.fd8dd34d4e169965eb0d.bundle.js:105058)
    at Object.mapper [as current] (vendors~main.fd8dd34d4e169965eb0d.bundle.js:105491)
    at ManagerConsumer (vendors~main.fd8dd34d4e169965eb0d.bundle.js:36887)
    at renderWithHooks (vendors~main.fd8dd34d4e169965eb0d.bundle.js:129672)
    at mountIndeterminateComponent (vendors~main.fd8dd34d4e169965eb0d.bundle.js:131906)
    at beginWork$1 (vendors~main.fd8dd34d4e169965eb0d.bundle.js:133050)
    at HTMLUnknownElement.callCallback (vendors~main.fd8dd34d4e169965eb0d.bundle.js:114911)
```
```
The above error occurred in the <LocationProvider> component:
    in LocationProvider (created by Root)
    in HelmetProvider (created by Root)
    in Root

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
```

I think that the problem is with integration between **emotion** and **storybook** after the upgrade by `sb upgrade` script.
The above errors are from level of **emotion** code.

My _package.json_:
```
    "devDependencies": {
        "emotion": "^10.0.27",
        "@storybook/addon-a11y": "^6.0.7",
        "@storybook/addon-actions": "^6.0.7",
        "@storybook/addon-links": "^6.0.7",
        "@storybook/addon-viewport": "^6.0.7",
        "@storybook/react": "^6.0.7",
        "@storybook/addon-essentials": "^6.0.7",
        "@storybook/node-logger": "^6.0.7",
        "@storybook/preset-create-react-app": "^3.1.4",
        "react": "^16.8.0",
    },
```

For me, solution was removing _/node_modules_ and _yarn.lock_ (and then installing everything again). I know that it is not professional solution, but it works.

**Comment by bradringel** [comment]
I'm running into this as well. I upgraded our storybook from 5.3 to 6 and started seeing this. I suspect that it has to do with conflicting versions of emotion, I know i've seen that happen before. I have version 10.0.22 of @emotion/core installed, 10.0.19 of emotion-theming and a decorator that looks like this:
```jsx
import { ThemeProvider } from 'emotion-theming';
import React from 'react';

import themes from '../../src/UI/themes';

const ThemeDecorator = Story => (
   <ThemeProvider theme={themes.herbert}>
      <Story />
   </ThemeProvider>
);

export default ThemeDecorator;
```

**Comment by philxavier** [comment]
Same here guys =0

**Comment by Nvveen** [comment]
This is an issue with multiple `@emotion/core` dependencies across the tree, which can be helped by forcing the version across all packages. Beware this might break any package using the older versions. In my case I was using `@chakra-ui/core` depending on `@emotion/core@10.0.35` while storybook was depending on `@emotion/core@10.0.28`, so (I'm hoping) not too big of a change.

To force the versions, either do `yarn install --flat` and choose the version of the package you want to use, or add a `resolutions` field to `package.json` with the package name and the required version, after which you should delete the lock-file and reinstall.

For more information, see https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-flat

**Comment by DominicTobias-b1** [comment]
> For me, solution was removing _/node_modules_ and _yarn.lock_ (and then installing everything again). I know that it is not professional solution, but it works.

Thanks that works, although simple stories are using so many resources to load in v6 it's all but crashing my 32GB MBP 2019 and the browser suggests closing the unresponsive page for a while 😅

**Comment by DominicTobias-b1** [comment]
> This is an issue with multiple `@emotion/core` dependencies across the tree, which can be helped by forcing the version across all packages. Beware this might break any package using the older versions. In my case I was using `@chakra-ui/core` depending on `@emotion/core@10.0.35` while storybook was depending on `@emotion/core@10.0.28`, so (I'm hoping) not too big of a change.
> 
> To force the versions, either do `yarn install --flat` and choose the version of the package you want to use, or add a `resolutions` field to `package.json` with the package name and the required version, after which you should delete the lock-file and reinstall.
> 
> For more information, see https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-flat

Very nice! I had to free up 18GB of RAM to get a story to load in under 10seconds in v6, even changing a button state takes 8 seconds, was working totally fine in v5. I started to suspect something was up with Emotion when I saw it was the bottleneck of every long running scripting operation (goes further down than screenshot shows these are just the top ones):

![Screenshot 2020-08-24 at 12 37 04 PM](https://user-images.githubusercontent.com/58683337/91004757-ae33a080-e607-11ea-85d3-4aaf8f921a20.png)

Sadly even pinning to 10.0.28 doesn't fix it.

**Comment by nelsonleite** [comment]
Following @Nvveen tip, I found [yarn-deduplicate](https://github.com/atlassian/yarn-deduplicate) and running `yarn-deduplicate --packages @emotion/core,@emotion/styled` actually worked! It removed the duplicates on my `yarn.lock` and it seems to be working just fine 🙂

**Comment by LoicGoyet** [comment]
> Following @Nvveen tip, I found [yarn-deduplicate](https://github.com/atlassian/yarn-deduplicate) and running `yarn-deduplicate --packages @emotion/core,@emotion/styled` actually worked! It removed the duplicates on my `yarn.lock` and it seems to be working just fine 🙂

Looks great, but I think we need to keep find out a solution that would be "dependencies manager agnostic". So we need to handle this dependency issue without specific npm or yarn fix.

**Comment by nscmnto** [comment]
> This is an issue with multiple `@emotion/core` dependencies across the tree, which can be helped by forcing the version across all packages. Beware this might break any package using the older versions. In my case I was using `@chakra-ui/core` depending on `@emotion/core@10.0.35` while storybook was depending on `@emotion/core@10.0.28`, so (I'm hoping) not too big of a change.
> 
> To force the versions, either do `yarn install --flat` and choose the version of the package you want to use, or add a `resolutions` field to `package.json` with the package name and the required version, after which you should delete the lock-file and reinstall.
> 
> For more information, see https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-flat

Thanks! In my case, running `yarn-deduplicate`, pointing `@emotion/core` to `10.0.28` in `package.json` `resolutions`, removing `node_modules` and then `yarn` seemed to work :)

**Comment by jackoliver** [comment]
For anyone trying to get this to work with Gatsby, @nscmnto and @Nvveen's solution works. :)

**Comment by deterralba** [comment]
For the npm team, using `npm dedupe` fixed it for me docs.npmjs.com/cli/dedupe

**Comment by anilanar** [comment]
@shilman What if I want to use a different version of emotion than the storybook packages use?
At @userlike, we use Emotion v11 pre-release.

**Comment by shilman** [comment]
@anilanar Storybook's manager (surrounding UI code) uses one version of emotion. Its preview (user code) doesn't use emotion at all, except if you're using `addon-docs` and in the docs tab. So if you want to use a different version of emotion, don't use `addon-docs` (you can disable it from essentials), and that might be all you need. If that doesn't work, perhaps there is some yarn or webpack resolution magic that forces the right version .. but I haven't tried that yet.

**Comment by Jimmydalecleveland** [comment]
I've tried to resolve this for quite some time, using the suggestions in this issue, with no success. I have a monorepo, which always makes things a little more complicated, especially with storybook historically for me.

I've narrowed it down to Emotion, no surprise there, with either `emotion-theming`, `@emotion/core`, or `@emotion/styled` being installed, all break my storybook with the same errors listed throughout this issue, even without using any existing stories I've written (just the ones from `sb init`).

I've tried deleting `yarn.lock`, deduplicating as suggested, an various other tinkerings to no avail. If I remove all those emotion packages, storybook runs correctly.

Mostly writing this to add another case to the problem and make it known, but if anyone has advice I will very much appreciate it.

**Comment by shilman** [comment]
For anybody who's experiencing this problem, what happens when you run Storybook with the `--no-dll` flag?

**Comment by SimonS** [comment]
> For anybody who's experiencing this problem, what happens when you run Storybook with the `--no-dll` flag?

@shilman that fixes it for me.

**Comment by shilman** [comment]
@SimonS documented the implications here: #12874 Please LMK if that needs more clarification!

**Comment by shilman** [comment]
Egads!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.1.0-alpha.28 containing PR #12874 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

Closing this issue. Please re-open if you think there's still more to do.

**Comment by shilman** [comment]
Whoopee!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.0.27 containing PR #12874 that references this issue. Upgrade today to the `@latest` NPM tag to try it out!

```
npx sb upgrade
```

**Comment by ayeletcohen** [comment]
Unfortunately I'm still facing that issue, even after upgrade to v6.0.27.

Tried upgrading my dependencies to versions required by storybook;
Tried `npm dedup`;
Tried removing node_modules and package-lock.json;
Tried installing storybook without installing my own npm dependencies.
But no luck, I'm still getting `Uncaught TypeError: Cannot read property 'fonts' of undefined`.

I used storybook in the past and would really love to introduce it to my new team, but I just can't find a way to install it.
Would appreciate any help on this. Thanks.

**Comment by shilman** [comment]
@ayeletcohen Did you try running Storybook with the `--no-dll` flag?

**Comment by ayeletcohen** [comment]
Yes, I have, not working.
I'm working on a monorepo solution, and succeeded installing storybook on the root level - but this is obviously not optimal. 
Thanks.

**Comment by shilman** [comment]
@ayeletcohen Glad you figured out a workaround. Hopefully we can come up with something better as part fo the 6.1 release.

**Comment by josiah-roberts** [comment]
> Unfortunately I'm still facing that issue, even after upgrade to v6.0.27.
> 
> Tried upgrading my dependencies to versions required by storybook;
> Tried `npm dedup`;
> Tried removing node_modules and package-lock.json;
> Tried installing storybook without installing my own npm dependencies.
> But no luck, I'm still getting `Uncaught TypeError: Cannot read property 'fonts' of undefined`.
> 
> I used storybook in the past and would really love to introduce it to my new team, but I just can't find a way to install it.
> Would appreciate any help on this. Thanks.

I'm getting the same symptoms here.

**Comment by choznerol** [comment]
Upgrade to >= `6.1.0-alpha.33` fix `Uncaught TypeError: can't access property "fonts", typography is undefined` for me.

With `6.1.0-alpha.28` and `6.1.0-alpha.29`, there is the same error:
> <img width="1102" alt="圖片" src="https://user-images.githubusercontent.com/12410942/97944810-a55a0a80-1dc0-11eb-9de2-85483b40c06b.png">

With `6.1.0-alpha.30`, `6.1.0-alpha.31` and `6.1.0-alpha.32`, it because another error:
> <img width="1139" alt="圖片" src="https://user-images.githubusercontent.com/12410942/97944800-9a06df00-1dc0-11eb-8c88-fd9cf8d7839f.png">

After upgrade to `6.1.0-alpha.33` the initial build is success. Mayby #12926 fixed this?

**Comment by amowu** [comment]
I tried to remove the `node_modules` and reinstall it, and it can runs.

**Comment by sarahbethfederman** [comment]
I'm having this issue currently, but only when running Storyshots. The test always fails with `TypeError: Cannot read property 'fonts' of undefined` but running storybook works fine

**Comment by shilman** [comment]
@sarahbethfederman are you using them in your stories or on your docs pages?

**Comment by sarahbethfederman** [comment]
Yes? Story in an MDX docs page 😅

**Comment by sarahbethfederman** [comment]
So seems I'm able to use the ColorPalette in the MDX doc outside of the story. I only get the issue when using it inside of a <Story /> block. I get the error in both Chromatic and Storyshots. 

For now I've disabled the story in both, but I'd really like to be able to snapshot the colors. Any ideas?

**Comment by shilman** [comment]
Storybook's [doc blocks](https://storybook.js.org/docs/react/writing-docs/doc-blocks) require a [theme context](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/blocks/DocsContainer.tsx#L83) to work. I think if you provide one in a decorator you should be fine. This is not really supported usage, however.

**Comment by sarahbethfederman** [comment]
Oh yeah I was trying to do it with just the ThemeProvider. Wrapping the decorator with `<DocsContainer context={context}>` instead fixed it. I get that its unsupported (though it is nice to be able to create a color palette story), are there any complications you see that might arise from consuming the docs context within the stories I should be aware of? I've hacked it so that it only wraps stories with the DocsContainer if they pass an `includeDocsContext` parameter

**Comment by shilman** [comment]
@sarahbethfederman it shouldn't be a problem. unsupported just means that we won't go out of our way to keep it working between minor version bumps and won't provide migration instructions on major bumps. i'm hoping we'll rework a bunch of this stuff later this year, and perhaps then we can better support your use case. 🤞

**Comment by simonhermann** [comment]
Upgrading from 7.0.21 to 7.0.27 (including all addons) fixed it for me 🤷

**Comment by TheColorRed** [comment]
For me, I had to add the following to my `devDependencies`:

```json
{
  "devDependencies": {
    "@emotion/react": "^11.10.4",
    "@emotion/styled": "^11.10.4",
  }
}
```

I am not sure why they weren't installing otherwise...

**Comment by kfayelun** [comment]
> @vamshi9666 I have seen this as well when trying to use `IconGallery` and `IconItem` inside a `stories.js`(seems to be fine inside `.mdx`) story. Was this your experience as well?

This issue has come back for us now: `Error: Cannot read properties of undefined (reading 'fonts')`. We've had it before, maybe a little less than a year ago, but I can't exactly remember what triggered it then. Updating the packages fixed the issue I think (back then).  

What happened now; I moved some code using `<IconGallery />` from an `.mdx`-file to a `.tsx`-file and this triggered the error, just like @vamshi9666 describes 3 years ago. So it works in MDX, but not in tsx. 
We just upgraded to `7.4.6` last week. I have deleted node modules, doesn't help. I'd like to avoid deleting `package-lock.json`. 

Screenshots of the mdx component that works, and the tsx component that fails, not sure how much value these adds though... I can't easily create a reproduction unfortunately. 

<img width="1796" alt="Screenshot 2023-10-11 at 12 59 58" src="https://github.com/storybookjs/storybook/assets/4848101/04a30096-ca8e-482f-92a0-9d6c8381363e">
<img width="1796" alt="Screenshot 2023-10-11 at 13 00 33" src="https://github.com/storybookjs/storybook/assets/4848101/5d1872ea-b331-4fc4-b87c-fc3c9b09ca1d">

From console:
```
index.mjs:176 Uncaught TypeError: Cannot read properties of undefined (reading 'fonts')
    at withReset (index.mjs:176:14652)
    at handleInterpolation (index.mjs:14:3193)
    at serializeStyles (index.mjs:19:1660)
    at index.mjs:24:1381
    at index.mjs:19:2916
    at renderWithHooks (react-dom.development.js:16305:18)
    at updateForwardRef (react-dom.development.js:19226:20)
    at beginWork (react-dom.development.js:21636:16)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
```
```
react-dom.development.js:18687 The above error occurred in the <Styled(div)> component:

    at http://localhost:6006/node_modules/.cache/sb-vite/deps/chunk-D2LYYWGF.js?v=d470fd51:1445:45
    at IconGallery (http://localhost:6006/node_modules/.cache/sb-vite/deps/chunk-SRLLPJOT.js?v=d470fd51:1713:22)
    at unboundStoryFn (http://localhost:6006/sb-preview/runtime.js:41:3330)
    at ErrorBoundary (http://localhost:6006/node_modules/.cache/sb-vite/deps/chunk-5QULKNVV.js?v=d470fd51:27:5)
    at WithCallback (http://localhost:6006/node_modules/.cache/sb-vite/deps/chunk-KRDZVWSL.js?v=d470fd51:15:23)
```

**Comment by enix79** [comment]
I also get this error message again. The strange thing is, that I can't find any occurrence of "@emotion" in the whole project, including node_modules...

**Comment by gehaktmolen** [comment]
I just had the aforementioned problem with the latest version of Storybook (7.5.2). Indeed no `@emotion` occurrences. However, installing `@storybook/theming` worked for me.

**Comment by simonhermann** [comment]
Thanks so much @gehaktmolen! installing @storybook/theming actually helps!

**Comment by enix79** [comment]
Hey @fastndead, make sure that all of your storybook dependencies have the same version and update them as a group only. This should fix the problem.

**Comment by fastndead** [comment]
@enix79 thanks for the advice! I had the same versions of the storybook packages in the package.json, but i had left `^` carets for them and yarn resolved them in all the different wrong ways. So i additional advice for people with that issue: **make sure your storybook dependencies have the same _fixed_ version**

**Comment by enix79** [comment]
@fastndead, yes, ultimately what counts is what you have in your package-lock.json. So look it up there to be 100%.

**Comment by tylercraft** [comment]
I just encountered this on `.mdx` files using `<ColorPalette />` tags. 

I went through all of our `package.json` updates to narrow down the culprit (though I'm sure it could be different for each instance). For us, it was updating `@types/react` from `18.2.66` to `18.2.67`. The only difference in our `yarn.lock` file is this:

Previously (working):
```
"@types/react@npm:18.2.66, @types/react@npm:^16.8.0 || ^17.0.0 || ^18.0.0":
```

With the update that breaks it:
```
"@types/react@npm:18.2.67":
  version: 18.2.67
  resolution: "@types/react@npm:18.2.67"
  dependencies:
    "@types/prop-types": "*"
    "@types/scheduler": "*"
    csstype: ^3.0.2
  checksum: 9e48122b0e8d0ae20601f613aca32289eeb94d55c58ff03a7dc532d80ed289a1ba9715f14b773bd926ccdd63e503717f14264143ef531dc370063abe09dee0b2
  languageName: node
  linkType: hard

"@types/react@npm:^16.8.0 || ^17.0.0 || ^18.0.0":
```

I'm not sure why a change in types would impact anything here.. but it does. I can consistently go back and forth on that version to break it, or have it work.

**Comment by peterschmiz** [comment]
Hi,

We have the same issue, i tried re-installing packages, deleted package-lock.json and played around with the versions, but I still have (and only for one story, where I use Typeset bloc) the error.

**Comment by kherock** [comment]
I'm also seeing this issue with a Yarn PnP and pnpm projects that have storybooks in a monorepo.

Installing `@storybook/theming` at the project root fixes the issue for me, which suggests that there is a missing peer dependency somewhere.

I also only experience this issue with the webpack builder -  the Vite builder works fine.

**Comment by tylercraft** [comment]
Sadly adding `@storybook/theming` did not fix it for me. We are using webpack as well. And while we are installing via yarn, we are not using PnP.

**Comment by kherock** [comment]
I've isolated the issue a bit further - I think there's a bug where things simply break if there are multiple instances of `@storybook/theming` in your dependency tree.

@tylercraft try using this in your `webpackFinal`!

```js
  webpackFinal(config) {
    Object.assign(config.resolve.alias, {
      ['@storybook/theming']: dirname(require.resolve('@storybook/theming/package.json'))
    })
    return config;
  },
```

**Comment by peterschmiz** [comment]
Adding the `@storbook/theming` solved the issue for us (we are using Vite)

**Comment by tylercraft** [comment]
> I've isolated the issue a bit further - I think there's a bug where things simply break if there are multiple instances of `@storybook/theming` in your dependency tree.
> 
> @tylercraft try using this in your `webpackFinal`!
> 
> ```js
>   webpackFinal(config) {
>     Object.assign(config.resolve.alias, {
>       '@storybook/theming': dirname(require.resolve('@storybook/theming/package.json'))
>     })
>     return config;
>   },
> ```

Thanks for looking into this @kherock. However, it does not remedy the issue for me. I've also gone through and removed all non essential storybook packages and tried to load, and no luck.

**Comment by valentinpalkovic** [comment]
If you still have issues, could you please open an new issue with a reproduction?

**Comment by shannonhochkins** [comment]
Ran into this when upgrading storybook from 7 -> 8, after hours of debugging, I found that this particular fonts property is attempting to retrieve via a `useTheme` hook, which is then used via the `@storybook/theming` package.

The way I solved this was to define the jsx decorator with the theme in the preview.tsx (or preview.js if not using typescript)

```
import { withThemeFromJSXProvider } from '@storybook/addon-themes';
import { ThemeProvider } from '@storybook/theming';

const THEME = {
  typography: {
    fonts: {
      base: 'Arial, sans-serif',
      mono: 'Courier, monospace'
    }
  }
};
export default {
  decorators: [
    withThemeFromJSXProvider({
      themes: {
        dark: THEME,
        light: THEME,
      },
      defaultTheme: 'dark',
      Provider: ThemeProvider,
    }),
  ]
}
```

All the docs say to import the ThemeProvider from @emotion/react which might be true for some use cases, but for the particular case of the Source component using the useTheme hook from the @storybook/theming package, this solved it!

**Comment by amcdnl** [comment]
<img width="1227" alt="image" src="https://github.com/storybookjs/storybook/assets/227909/2fda48ff-6b02-4e3a-8880-891ff687473c">

I'm also running into this using the following deps and a brand new app.

```
    "@storybook/addon-essentials": "^8.1.10",
    "@storybook/blocks": "^8.1.10",
    "@storybook/manager-api": "^8.1.10",
    "@storybook/react": "^8.1.10",
    "@storybook/react-vite": "^8.1.10",
    "@storybook/test": "^8.1.10",
    "@storybook/theming": "^8.1.10",
    "storybook": "^8.1.10",
```

**Comment by AnsonH** [comment]
> [!WARNING]
> TL;DR: Do NOT use [Source](https://storybook.js.org/docs/writing-docs/doc-blocks#source) block inside a **non-MDX storybook file** (i.e. `**.stories.(js|jsx|ts|tsx)`). See [this workaround](https://github.com/storybookjs/storybook/issues/12016#issuecomment-2076905737) if you really need to.

I faced this bug when trying to use [Source](https://storybook.js.org/docs/writing-docs/doc-blocks#source) block inside a non-MDX file of `**.stories.tsx`. 

However, the [documentation](https://storybook.js.org/docs/writing-docs/doc-blocks#why-cant-i-use-the-doc-blocks-inside-my-stories) states that Doc Blocks are primarily designed for MDX files, and adding Doc Blocks to non-MDXC storybook files will cause errors. 

Looking at `Source` block's source code, it calls `useTheme()` to get the `typography` property ([code](https://github.com/storybookjs/storybook/blob/v8.1.10/code/ui/blocks/src/components/Source.tsx#L104)), then it accesses `typography.fonts.XXX` ([code](https://github.com/storybookjs/storybook/blob/v8.1.10/code/ui/blocks/src/components/Source.tsx#L132-L133)). My guess is that stories generated from MDX files has a custom Emotion `ThemeProvider` to provide the typography themes, and this theme provider is absent in non-MDX storybook files. That's because `Source` is not supposed to be used outside non-MDX files.

If you really want to use `Source` in your `.stories.ts(x)` files, [this workaround](https://github.com/storybookjs/storybook/issues/12016#issuecomment-2076905737) worked for me (remember to install the dev dependency `@storybook/addon-themes`).

**Comment by GaEhrlich127** [comment]
> [!WARNING]
> All Doc Blocks (ArgTypes, Canvas, ColorPalette, Controls, Description, IconGallery, Markdown, Meta, Primary, Source, Stories, Story, Subtitle, Title, Typeset, Unstyled) are not supported in non `.mdx` stories.

[This warning directly above me](https://github.com/storybookjs/storybook/issues/12016#issuecomment-2191359084) is a little bit too specific, especially if someone (like myself) is googling an error and somewhat skimming to look for the specific case they're dealing with. 

The rest of the comment is excellent and does well at documenting the issue, Although note that in Storybook 9, `@storybook/theming` was replaced with `storybook/theming`. The fix they provided is ultimately still the idea that worked for me, however, did not solve my problem with IconGallery specifically.

A more general fix that appears to be working well for me in Storybook 10 is importing and providing the default themes found at [`storybook/code/core/src/theming/base.ts`](https://github.com/storybookjs/storybook/blob/13b7cffdc0a80b9a44314e4cf84c191e825ecb20/code/core/src/theming/base.ts):
```
import { ThemeProvider } from 'storybook/theming';
import { background, color, typography } from '@storybook/theming';

const theme = {
  typography,
  background,
  color,
};

export default {
  decorators: [(Story: any) => (
    {/* @ts-ignore */}
    <ThemeProvider theme={theme}>
      <Story />
    </ThemeProvider>
)],
```

All that said and done, you really probably shouldn't go through this effort just to make an unsupported feature work, since it's not supported in the first place...
